workspace/src/github.com/darkcrux/go-ceph-rest-api/handler_pool.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"fmt"
+
+	"net/http"
+
+	"encoding/json"
+	"github.com/gorilla/mux"
+)
+
+const (
+	// actions
+	create = "create"
+	// get = "get"
+	// delete = "delete"
+
+	// methods
+	// get = "GET"
+	// post = "POST"
+	put = "PUT"
+	// delete = "DELETE"
+)
+
+type poolAPI struct {
+	Res    http.ResponseWriter
+	Req    *http.Request
+	Params map[string]string
+}
+
+type Response struct {
+	Status string      `json:"status"`
+	Output interface{} `json:"output"`
+}
+
+func poolHandler(res http.ResponseWriter, req *http.Request) {
+	pool := poolAPI{res, req, mux.Vars(req)}
+	callback := func(resp Response) {
+		json_response, _ := json.Marshal(resp)
+		fmt.Fprint(res, string(json_response))
+	}
+	pool.handlePut(callback)
+	// pool.handlePost(callback)
+	// pool.handleGet(callback)
+	// pool.handleDelete(callback)
+}
```

Comment:
we can have a switch here:

```
switch pool.Params["action"] {
case ActionPut: ...
}
```

ugly switches. haha. Umm.. or any other cleaner way of doing it?